src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -4,192 +4,177 @@
+			setDefaultServerInitProperties(server);
+
+			foreach (SMO.Database db in server.Databases)
+			{
+				results.Add(CreateItem(db));
+			}
+
+			results = results.Where(s => SystemDatabases.Any(x => this.NameMatches(x, s))).Concat(
+						 results.Where(s => SystemDatabases.All(x => !this.NameMatches(x, s)))).ToList();
+
+			SetResponse(response, results.ToArray());
+			serverConnection.Disconnect();
+
+			return response;
+		}
+
+		protected abstract void setDefaultServerInitProperties(SMO.Server server);
+		protected abstract bool NameMatches(string databaseName, T item);
+		protected abstract T CreateItem(SMO.Database database);
+		protected abstract void SetResponse(ListDatabasesResponse response, T[] results);
+	}
+
+	/// <summary>
+	/// database names handler
+	/// </summary>
+	class DatabaseNamesHandler : ListDatabaseRequestHandler<string>
+	{
+		protected override void setDefaultServerInitProperties(SMO.Server server)
+		{
+			server.SetDefaultInitFields(typeof(Database), new[] { nameof(Database.Name) });
+		}
+
+		protected override string CreateItem(SMO.Database database)
+		{
+			return database.Name;
+		}
+
+		protected override bool NameMatches(string databaseName, string item)
+		{
+			return databaseName == item;
+		}
+
+		protected override void SetResponse(ListDatabasesResponse response, string[] results)
+		{
+			response.DatabaseNames = results;
+		}
+	}
+
+
+	abstract class BaseDatabaseDetailHandler : ListDatabaseRequestHandler<DatabaseInfo>
+	{
+		protected override void setDefaultServerInitProperties(SMO.Server server)
+		{
+			server.SetDefaultInitFields(typeof(Database), new[] { nameof(Database.Name), nameof(Database.State), nameof(Database.Size) });
+		}
+		protected override bool NameMatches(string databaseName, DatabaseInfo item)
+		{
+			return databaseName == item.Options[ListDatabasesRequestDatabaseProperties.Name].ToString();
+		}
+
+		protected override void SetResponse(ListDatabasesResponse response, DatabaseInfo[] results)
+		{
+			response.Databases = results;
+		}
+
+		protected override DatabaseInfo CreateItem(SMO.Database database)
+		{
+			DatabaseInfo databaseInfo = new DatabaseInfo();
+			SetProperties(database, databaseInfo);
+			return databaseInfo;
+		}
+
+		protected virtual void SetProperties(SMO.Database database, DatabaseInfo databaseInfo)
+		{
+			databaseInfo.Options[ListDatabasesRequestDatabaseProperties.Name] = database.Name;
+			databaseInfo.Options[ListDatabasesRequestDatabaseProperties.State] = database.State.ToString();
+			databaseInfo.Options[ListDatabasesRequestDatabaseProperties.SizeInMB] = database.Size.ToString();
+		}
+	}
+
+	/// <summary>
+	/// Standalone SQL Server database detail handler
+	/// </summary>
+	class SqlServerDatabaseDetailHandler : BaseDatabaseDetailHandler
+	{
+		protected override void setDefaultServerInitProperties(SMO.Server server)
+		{
+			server.SetDefaultInitFields(typeof(Database), new[] { nameof(Database.LastBackupDate) });
```

Comment:
call base.set too

Reply:
I guess, I won't have to do this if I'm using overridden properties instead of this method.